### PR TITLE
Logger in signatures

### PR DESCRIPTION
**Pull request by NDStrahilevitz**
### 1. Explain what the PR does

1.  logger: change method names to .Levelw
    
    Previously logger methods were of format .Level, which didn't reflect the
    internal usage of wide printing (.Levelw).
    Change is done to reflect that and later allow addition of the other
    formats (.Level and .Levelf)

2. types: add SignatureContext type for init
    
    Add a Context type for signatures in order to have a centralized struct
    for adding context injections to signatures.
    This change is included with the addition of a logger to the context.

3. signatures: signatures: use new SignatureContext
    
    Update all included signatures to use the SignatureContext type.
    Also introduce logger example in docs.

### 2. Explain how to test it

Add a logger usage in one of the provided signatures compile and trigger it (Easiest one to test is ptrace_code_injection).

### CODE REVIEW

File: go.mod
Comment (NDStrahilevitz) [inline go.mod:114]: After final approval, split the PR and change this to actual commit bump

**Comment by rafaeldtinoco** [comment]
Sorry for the noise, we were reaching docker hub rate limits and I tried something that did not work. Could you please rebase and try the tests again ? Thanks!

**Comment by rafaeldtinoco** [comment]
Adding to v0.13.0 milestone as this is chore and does not have an associated issue.

File: pkg/signatures/engine/engine.go
Comment (geyslan) [inline pkg/signatures/engine/engine.go:248]: Couldn't the signature package just import the logger and use it at the package level?

File: pkg/logger/logger.go
Comment (geyslan) [inline pkg/logger/logger.go:574]: If the question below is true, this is not required.

File: docs/classic-docs/docs/detecting/golang.md
Comment (geyslan) [inline docs/classic-docs/docs/detecting/golang.md:70]: It should be this?

```suggestion
                    sig.logger.Debugw("no arguments found")
```

File: pkg/signatures/engine/engine.go
Comment (geyslan) [reply pkg/signatures/engine/engine.go:248]: Ah never-mind. I got the need.

File: pkg/logger/logger.go
Comment (geyslan) [reply pkg/logger/logger.go:574]: Perhaps a comment letting the caller know that it's receiving a pointer so direct changes to the returned logger will affect the whole system.

File: types/detect/detect.go
Comment (geyslan) [inline types/detect/detect.go:69]: This answers my previous questions. Awesome.

**Review by geyslan — APPROVED** [review]
LGTM

**Comment by NDStrahilevitz** [comment]
I will be splitting this up to the types PR tomorrow and then merge this once I fix the types to the commit.

**Pull request merged**
